### PR TITLE
fix:FPLA-2273: Remove defensive code/toggle for FPLA-2202

### PR DESCRIPTION
**Pull request by samuelzerozero**
**Before creating a pull request make sure that:**

- [x ] PR's title includes JIRA ticket number and follow [Conventional Commits](https://www.conventionalcommits.org/) specification
- [ x] README and other documentation has been updated / added (if needed)
- [ x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/FPLA-2273

### Change description ###
Cleanup job after the go live of FPLA-2202

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by tomaszpowroznik — APPROVED** [review]

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderController.java [hunk 1]
```diff
@@ -43,11 +41,9 @@ public AboutToStartOrSubmitCallbackResponse handleMidEventValidateDateOfIssue(
         List<String> errors = validateGroupService.validateGroup(caseData, DateOfIssueGroup.class);
 
         if (GeneratedOrderSubtype.INTERIM.equals(caseData.getOrderTypeAndDocument().getSubtype())) {
-            if (caseData.getInterimEndDate() != null) { // TODO: IF To be removed after FPLA-2202 goes live
-                errors.addAll(validateGroupService.validateGroup(
-                    caseData.getInterimEndDate(), InterimEndDateGroup.class
-                ));
-            }
+            errors.addAll(validateGroupService.validateGroup(
```
Comment (lewisbirks) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderController.java:44]: Do you need the condition that was removed from the other method?
```java
        if (interimEndDate.getType() == NAMED_DATE) {	
```
although maybe it is now NAMED_DATE and SPECIFIC_TIME_NAMED_DATE.
Cannot remember how nulls interact with the validation

Comment (tomaszpowroznik) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderController.java:44]: InterimEndDate uses @Future validators

https://docs.oracle.com/javaee/6/api/javax/validation/constraints/Future.html

nulls elements are considered valid.

Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderController.java:44]: Missed the null elements part...
I need more sleep 🙂

Comment (samuelzerozero) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderController.java:44]: I've added some more tests that actually cover the case for time on the top of dates to make it more explicit. Doing that I probably figured out why those tests were not there 😄  Basically the LocalValidatorFactoryBean by default takes its own clock provider. That caused to have the test clock to be at fixed time, and the validator with the standard one. That meant flakiness if the test would run let say at minute 11:21, priming the data at that instant, and then running the validation at 11:22.

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/FixedTimeConfiguration.java [hunk 2]
```diff
@@ -5,19 +5,37 @@
...
-import java.time.LocalDateTime;
+import java.time.Clock;
 import java.time.ZonedDateTime;
+import javax.validation.ClockProvider;
 
 @Configuration
 @Import(TestConfiguration.class)
 public class FixedTimeConfiguration {
-    public static final LocalDateTime NOW = ZonedDateTime.now().toLocalDateTime();
+    public static final ZonedDateTime NOW = ZonedDateTime.now();
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/utils/FixedTimeConfiguration.java:18]: Not sure that this should be public

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderController.java (hunk 1 above)
Comment (lewisbirks) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderController.java:44]: I love time, always a joy to test 😅

File: service/src/test/java/uk/gov/hmcts/reform/fpl/utils/FixedTimeConfiguration.java (hunk 2 above)
Comment (samuelzerozero) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/utils/FixedTimeConfiguration.java]: changed to private

**Review by lewisbirks — APPROVED** [review]

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderControllerTest.java [hunk 3]
```diff
@@ -49,58 +48,81 @@
 
     @Test
     void shouldReturnErrorsWhenTheDateOfIssueIsInFuture() {
-        CaseDetails caseDetails = createCaseDetails(PREVENT_REMOVAL, time.now().plusDays(1),
-            NAMED_DATE, ORDER_TYPE_AND_DOCUMENT, time.now());
+        CaseDetails caseDetails = createCaseDetails(PREVENT_REMOVAL, time.now().plusDays(1), ORDER_TYPE_AND_DOCUMENT);
```
Comment (lewisbirks) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderControllerTest.java:51]: minor for most of the test changes you have made here but not something you have done specifically. I do not believe you need to autowire in Time to this test, the base class `AbstractControllerTest` has it and also some convenience methods to get the time in LocalDate or LocalDateTime format

**Review by lewisbirks — APPROVED** [review]

**Review by lewisbirks — APPROVED** [review]

**Pull request merged**